src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs

```diff
@@ -27,16 +31,27 @@ public class ConsoleLoggerProvider : ILoggerProvider, ISupportExternalScope
         /// </summary>
         /// <param name="options">The options to create <see cref="ConsoleLogger"/> instances with.</param>
         public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options)
+            : this(options, Enumerable.Empty<IConsoleLogFormatter>()) { }
+
+        /// <summary>
+        /// Creates an instance of <see cref="ConsoleLoggerProvider"/>.
+        /// </summary>
+        /// <param name="options">The options to create <see cref="ConsoleLogger"/> instances with.</param>
+        /// <param name="formatters">Log formatters added for <see cref="ConsoleLogger"/> insteaces.</param>
+        public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options, IEnumerable<IConsoleLogFormatter> formatters)
         {
             _options = options;
             _loggers = new ConcurrentDictionary<string, ConsoleLogger>();
 
+            _formatters = SetFormatters(formatters);
+
             ReloadLoggerOptions(options.CurrentValue);
             _optionsReloadToken = _options.OnChange(ReloadLoggerOptions);
 
             _messageQueue = new ConsoleLoggerProcessor();
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
+                // [TODO]: check if VT enabled on windows, console mode.
```

Comment:
- [ ] For ANSI escapes, rather than adding check for VT enabled: would need to call SetConsoleMode and pass in ENABLE_VIRTUAL_TERMINAL_PROCESSING. If the function returns a success code, Windows supports it. If the function doesn't succeed, then it doesn't.

Reply:
Is it recommended to call SetConsoleMode  than just checking the mode using GetConsoleMode? does the Console logger is the one create the console session to be allowed to change the mode?

---
In reply to: [447632851](https://github.com/dotnet/runtime/pull/38616#discussion_r447632851) [](ancestors = 447632851)

Reply:
The console logger shouldn't side effect global process state like that.